utils/comma-separated-list.ts

```diff
@@ -0,0 +1,28 @@
+import * as t from "io-ts";
+
+/**
+ * Create a decoder that parses a list of comma-separated elements into an array of typed items, using the provided decoder
+ *
+ * @param decoder a io-ts decoder
+ *
+ * @returns either a decode error or the array of decoded items
+ */
+// eslint-disable-next-line @typescript-eslint/explicit-function-return-type
+export const CommaSeparatedListOf = (decoder: t.Mixed) =>
+  new t.Type<ReadonlyArray<t.TypeOf<typeof decoder>>, string, unknown>(
+    `CommaSeparatedListOf<${decoder.name}>`,
+    (value: unknown): value is ReadonlyArray<t.TypeOf<typeof decoder>> =>
+      Array.isArray(value) && value.every(e => decoder.is(e)),
+    (input /* , context */) =>
+      t.readonlyArray(decoder).decode(
+        typeof input === "string"
+          ? input
+              .split(",")
+              .map(e => e.trim())
+              .filter(Boolean)
+          : !input
+          ? [] // fallback to empty array in case of empty input
+          : input // it should not happen, but in case we let the decoder fail
+      ),
```

Comment:
This is not going to fail in case of missing env variable, I suggest to use this code instead:

```suggestion
      (input, context) =>
      typeof input === "undefined"
        ? t.failure(input, context, "input is undefined")
        : t.readonlyArray(decoder).decode(
            typeof input === "string"
              ? input
                  .split(",")
                  .map(e => e.trim())
                  .filter(Boolean)
              : !input
              ? [] // fallback to empty array in case of empty input
              : input // it should not happen, but in case we let the decoder fail
          ),
```

Reply:
Well, to ensure the set is not empty is not in the scope of this codec. If we need it (and I argue this is not the case), we can _refine_ it (using either `pipe` or `intersection`) just where the codeec is used

Reply:
Ok, whatever we are not handling this case right now